### PR TITLE
Fix global pool

### PR DESCRIPTION
**Pull request by jreidinger**
tested manually

### CODE REVIEW

File: src/include/dhcp-server/dialogs.rb [hunk 1]
```diff
@@ -378,27 +378,27 @@ def SectionTypeChoose
...
 
         par = DhcpServer.GetEntryParent(par_type, par_id)
         par_type = Ops.get(par, "type", "")
         par_id = Ops.get(par, "id", "")
       end
       possible = ["subnet", "host", "shared-network", "group", "pool", "class"]
-      if Builtins.contains(parents, "class") ||
-          Builtins.contains(parents, "host") ||
-          Builtins.contains(parents, "pool")
+      if ["class", "host", "pool"].include?(parents)
```
Comment (mchf) [inline src/include/dhcp-server/dialogs.rb]: does it really work as intended?

File: src/include/dhcp-server/dialogs.rb [hunk 2]
```diff
@@ -378,27 +378,27 @@ def SectionTypeChoose
...
-      f = []
-      if Builtins.contains(parents, "subnet")
-        f = Builtins.add(f, "subnet")
-        f = Builtins.add(f, "shared-network")
+      filtered = ["pool"]
+      if parents.include?("subnet")
+        filtered.delete("pool")
+        filtered << "subnet"
+        filtered << "shared-network"
+        filtered << "shared-network"
```
Comment (mchf) [inline src/include/dhcp-server/dialogs.rb]: why having ```filtered << "shared-network"``` twice?

File: src/include/dhcp-server/dialogs.rb (hunk 1 above)
Comment (mvidner) [inline src/include/dhcp-server/dialogs.rb]: I also think this is wrong

Comment (jreidinger) [inline src/include/dhcp-server/dialogs.rb]: you mean if this code do same think as before? yes. Logic there is that it check if in parents are leaps, which cannot have anything under itself....ah, parents can be also array, let me test. we in fact need intersect here

Comment (jreidinger) [inline src/include/dhcp-server/dialogs.rb]: I fixed it, but it is interesting, as in fact it is disabled somewhere in UI, so this button cannot be even clicked.

File: src/include/dhcp-server/dialogs.rb (hunk 2 above)
Comment (jreidinger) [inline src/include/dhcp-server/dialogs.rb]: thanks for catch, typo

File: src/include/dhcp-server/dialogs.rb (hunk 1 above)
Comment (jreidinger) [inline src/include/dhcp-server/dialogs.rb]: so my manual testing do not discover it.

**Comment by mchf** [comment]
lgtm

File: src/include/dhcp-server/dialogs.rb [hunk 3]
```diff
@@ -378,27 +378,26 @@ def SectionTypeChoose
...
-          Builtins.contains(parents, "host") ||
-          Builtins.contains(parents, "pool")
+      if ["class", "host", "pool"].include?(parents)
         return :back
       end
-      f = []
-      if Builtins.contains(parents, "subnet")
-        f = Builtins.add(f, "subnet")
-        f = Builtins.add(f, "shared-network")
+      filtered = []
```
Comment (jreidinger) [inline src/include/dhcp-server/dialogs.rb]: make sense, naming is always challenge...and previous `f` was horrible, it took me 5 minutes, just to recognize how it works, as old filer method and small `!` cause I think it is another allowed list.

File: package/yast2-dhcp-server.changes [hunk 4]
```diff
@@ -1,4 +1,11 @@
 -------------------------------------------------------------------
+Thu Oct  6 11:25:00 UTC 2016 - jreidinger@suse.com
+
+- prevent adding "pool" to global scope as it is allowed only in
+  network scope (bsc#937492)
```
Comment (mvidner) [inline package/yast2-dhcp-server.changes]: bsc -= 200000

**Pull request merged**